lib/spack/spack/cmd/buildcache.py

```diff
@@ -1143,22 +1143,25 @@ def update_index(mirror: spack.mirror.Mirror, update_keys=False):
 
     # Otherwise, assume a normal mirror.
     url = mirror.push_url
+    try:
+        bindist.generate_package_index(url_util.join(url, bindist.build_cache_relative_path()))
 
-    bindist.generate_package_index(url_util.join(url, bindist.build_cache_relative_path()))
-
-    if update_keys:
-        keys_url = url_util.join(
-            url, bindist.build_cache_relative_path(), bindist.build_cache_keys_relative_path()
-        )
+        if update_keys:
+            keys_url = url_util.join(
+                url, bindist.build_cache_relative_path(), bindist.build_cache_keys_relative_path()
+            )
 
-        bindist.generate_key_index(keys_url)
+            bindist.generate_key_index(keys_url)
+    except bindist.GenerateIndexError as e:
+        tty.error(str(e))
```

Comment:
Why are you catching this error at all?

Reply:
All `SpackError` are handled, much better than `tty.error`. Catching and logging it makes `spack --backtrace/--debug` useless.

Reply:
I guess you wrote this to fix a test from failing?

Did you look at why it failed at all?

Reply:
> Updating keys for an unsigned build cache doesn't make sense to me.

Except that CI is an instance that effectively runs `spack update-index --keys` whether the build cache is signed or not, which drives the point home that that's a valid use case, and it would be pedantic to differentiate between an empty directory (fine) and a non-existing directory (error). Just don't error.